src/coreComponents/physicsSolvers/wavePropagation/WaveSolverBase.hpp

```diff
@@ -154,6 +173,26 @@ class WaveSolverBase : public SolverBase
   /// Amount of seismoTrace that will be recorded for each receiver
   localIndex m_nsamplesSeismoTrace;
 
+  /// Mininum (x,y,z) coordinates of inner PML boundaries
+  R1Tensor m_xMinPML;
+
+  /// Maximum (x,y,z) coordinates of inner PML boundaries
+  R1Tensor m_xMaxPML;
+
+  /// Desired reflectivity of the PML region, used to compute the damping profile
+  real64 m_reflectivityPML;
+
+  /// Thickness of the PML region, used to compute the damping profile
+  R1Tensor m_thicknessMinXYZPML;
+  R1Tensor m_thicknessMaxXYZPML;
+
+  /// Wave speed in the PML region, used to compute the damping profile
+  R1Tensor m_waveSpeedMinXYZPML;
+  R1Tensor m_waveSpeedMaxXYZPML;
```

Comment:
To avoid having these data members remaining uninitialized in case the solver is used without PML, you could do the following:

1) You remove these data members from the solver

2) Instead, in the solver, you define a `struct` `PMLParameters` like this:
```
struct PMLParameters
{
  /// Mininum (x,y,z) coordinates of inner PML boundaries
  R1Tensor xMinPML;

  /// Maximum (x,y,z) coordinates of inner PML boundaries
  R1Tensor xMaxPML;

  /// Desired reflectivity of the PML region, used to compute the damping profile
  real64 reflectivityPML;

  /// Thickness of the PML region, used to compute the damping profile
  real64 thicknessMinXYZPML[3];
  real64 thicknessMaxXYZPML[3];

  /// Wave speed in the PML region, used to compute the damping profile
  real64 waveSpeedMinXYZPML[3];
  real64 waveSpeedMaxXYZPML[3];
};
```

3. In `initializePML`, since you know that PML is used, you can do: 
```
this->registerWrapper< PMLParameters >( viewKeyStruct::PMLParametersString() ).
      setRestartFlags( RestartFlags::NO_WRITE );
```

4. Then, every time you need to access the PML parameters, you can just do:
```
PMLParameters const & param = getReference< PMLParameters >( viewKeyStruct::PMLParametersString() );
```
and if someone ever tries to access `PMLParameters` in a simulation that does not have PML, a nice error message will be thrown.

5. In the PML pressure update, you can split the update in two kernels and only pass the PML parameters to the PML kernel

Reply:
Done. Your suggestion worked just fine. Thanks!